audio-echo/app/src/main/java/com/google/sample/echo/MainActivity.java

```diff
@@ -80,7 +89,16 @@ public boolean onOptionsItemSelected(MenuItem item) {
     }
 
     public void startEcho(View view) {
-        status_view.setText("StartCapture Button Clicked\n");
+        int permissionStatus = ActivityCompat.checkSelfPermission(this,
```

Comment:
yeah: the only reason I did it is because of the 80 character/line code review requirement which ,I think, is one of the dumbest things in the world :-(.  I am glad you pointed it out, thanks!